### DISCUSSION TITLE
uploading reading a data file (spike prime or RI hub)

### DISCUSSION THREAD

**Discussion by MichaelVerschaeve** [pybricks/support#1296] [Q&A]
How would one go about uploading a text file with data using the web development tool and reading that data in your python script?
The data file would be a text file of about 6 kB, 1200 lines of data which I would need to be able to seek in randomly (i.e. read any of the 1200 lines)

**Comment by dlech** [comment]
You can enclose the text in a multi-line string in a python file.

e.g. `text.py`

```python
TEXT = """line 1
line 2
...
"""
```

Then in your "real" Python script...
```python
from uio import StringIO

from text import TEXT

lines = StringIO(TEXT).readlines()

# lines should be a list of strings containing each line (including newline)
```
